src/core/qgsvectorlayerserverproperties.h

```diff
@@ -0,0 +1,142 @@
+
+      /**
+       * Constructor for WmsDimensionInfo.
+       */
+      explicit WmsDimensionInfo( const QString &dimName,
+                                 const QString &dimFieldName,
+                                 const QString &dimEndFieldName = QString(),
+                                 const QString &dimUnits = QString(),
+                                 const QString &dimUnitSymbol = QString(),
+                                 const int &dimDefaultDisplayType = QgsVectorLayerServerProperties::AllDimensionValues,
+                                 const QVariant &dimReferenceValue = QVariant() )
+        : name( dimName )
+        , fieldName( dimFieldName )
+        , endFieldName( dimEndFieldName )
+        , units( dimUnits )
+        , unitSymbol( dimUnitSymbol )
+        , defaultDisplayType( dimDefaultDisplayType )
+        , referenceValue( dimReferenceValue )
+      {}
+      QString name;
+      QString fieldName;
+      QString endFieldName;
+      QString units;
+      QString unitSymbol;
+      int defaultDisplayType;
+      QVariant referenceValue;
+    };
+
+    QgsVectorLayerServerProperties( QgsVectorLayer *layer = nullptr );
+
+    /**
+     * Returns predefined WMS Dimension names
+     * \since QGIS 3.10
+    */
+    const QStringList predefinedWmsDimensionNames() const { return mPredefinedWmsDimNames; }
+
+    /**
+     * Returns WMS Dimension default display labels
+     * \since QGIS 3.10
+    */
+    const QStringList wmsDimensionDefaultDisplayLabels() const { return mWmsDimDefaultDisplayLabels; }
+
+    /**
+     * Add a QGIS Server WMS Dimension
+     * \param wmsDimInfo QGIS Server WMS Dimension object with, name, field, etc
+     * \returns TRUE if QGIS Server WMS Dimension has been successfully added
+     * \since QGIS 3.10
+    */
+    bool addWmsDimension( const QgsVectorLayerServerProperties::WmsDimensionInfo &wmsDimInfo );
+
+    /**
+     * Removes a QGIS Server WMS Dimension
+     * \returns TRUE if QGIS Server WMS Dimension was found and successfully removed
+     * \since QGIS 3.10
+    */
+    bool removeWmsDimension( const QString &wmsDimName );
+
+    /**
+     * Returns the QGIS Server WMS Dimension list.
+     * \since QGIS 3.10
+     */
+    const QList<QgsVectorLayerServerProperties::WmsDimensionInfo> wmsDimensions() const;
+
+
+    /**
+     * Saves server properties to xml under the layer node
+     * \since QGIS 3.10
+     */
+    void writeXml( QDomNode &layer_node, QDomDocument &document ) const;
+
+    /**
+     * Reads server properties from project file.
+     * \since QGIS 3.10
+     */
+    void readXml( const QDomNode &layer_node );
+
+  private:                       // Private attributes
+
+    QgsVectorLayer *mLayer = nullptr;
+
+    //!stores QGIS Server WMS Dimension definitions
+    QList<QgsVectorLayerServerProperties::WmsDimensionInfo> mWmsDimensions;
+
+    //! Predefined QGIS Server WMS Dimension names
+    QStringList mPredefinedWmsDimNames { "Time", "Elevation" };
```

Comment:
Could we define an enum instead of using string values?

Reply:
I have use a static method to provide predefined WMS Dimension.